edgepi-leds.js

```diff
@@ -1,51 +1,54 @@
-module.exports = function(RED) {
-    const rpc = require("@edgepi-cloud/edgepi-rpc")
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
 
-    function LEDNode(config) {
-        // Create node with user config
-        RED.nodes.createNode(this, config);
-        const node = this;
-        const ipc_transport = "ipc:///tmp/edgepi.pipe"
-        const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
-        const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
+  function LEDNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
 
-        // Init new led
-        const led = new rpc.LEDService(transport);
-        if (led){
-            console.debug("LED node initialized on:", transport);
-            node.status({fill:"green", shape:"ring", text:"led initialized"});
+    setInitialConfigs(config).then((led) => {
+      node.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input received" });
+        try {
+          const state = msg.payload === "on" ? "turnOn" : "turnOff";
+          const ledPin = "LED" + msg.pin.toString();
```

Comment:
how does js behaves if msg.pin is null ?

Reply:
gives this error response in node-red output:

payload: "Error: Error 1: Invalid method input from client. The message used for the requested method input is either not the correct message type or is missing required fields."

Reply:
well in that case you should first to check if `msg.pin` is defined or not